sway-ir/tests/inline/int_to_ptr.ir

```diff
@@ -0,0 +1,25 @@
+// regex: VAR=v\d+
+// regex: LABEL=[[:alpha:]0-9]+:
+
+script {
+    fn a(b: u64) -> b256 {
+        entry:
+        v1 = int_to_ptr b to b256
+        ret b256 v1
+    }
+
+// check: fn main
+    fn main() -> b256 {
+        entry:
+
+// check: $(arg0=$VAR) = const u64 11
+        v0 = const u64 11
+// check: $(arg1=$VAR) = int_to_ptr $arg0 to b256
+// not: call
+        v1 = call a(v0)
```

Comment:
I know this is from the other PR but looking at it here I think these tests can be reorganised a bit.  The `inline` tests should really only be testing the inlining pass (and regressions), and we don't need to add a new test for every new opcode that we add.

The Sway `ir_generation` tests can verify that the right instruction is generated from the Sway source (e.g., an `int_to_ptr` pops out and not a `call`), and those tests also do the print -> parse -> print -> compare.  But we should be adding _something_ here for new opcodes.

I just took a peek at the `sway-ir/tests` dir and there _is_ already a `serialize` directory there..!  I forgot.  So that's where we should put tests for coverage in general.  It might need a bit of TLC, I'm not sure it's parsing it's own output like the `ir_generation` tests are.

Reply:
Yeah the test would have to change so I created a separate issue here: https://github.com/FuelLabs/sway/issues/2348